ambry-named-mysql/src/main/java/com/github/ambry/named/MySqlNamedBlobDb.java

```diff
@@ -241,21 +241,23 @@ public CompletableFuture<Page<NamedBlobRecord>> list(String accountName, String
   }
 
   @Override
-  public CompletableFuture<PutResult> put(NamedBlobRecord record, NamedBlobState state) {
+  public CompletableFuture<PutResult> put(NamedBlobRecord record, NamedBlobState state, Boolean isUpsert) {
     return executeTransactionAsync(record.getAccountName(), record.getContainerName(), true,
         (accountId, containerId, connection) -> {
           long startTime = System.currentTimeMillis();
-          NamedBlobRecord recordCurrent = null;
-          try {
-            recordCurrent = run_get_v2(record.getAccountName(), record.getContainerName(), record.getBlobName(),
-                GetOption.None, accountId, containerId, connection);
-          } catch (RestServiceException e) {
-            logger.trace("Skip exception in pulling data from db: accountId='{}', containerId='{}', blobName='{}': {}",
-                accountId, containerId, record.getBlobName(), e);
-          }
-          if (recordCurrent != null) {
-            throw buildException("PUT: Blob still alive", RestServiceErrorCode.Conflict, record.getAccountName(),
-                record.getContainerName(), record.getBlobName());
+          if (!(config.dbRelyOnNewTable && isUpsert)) {
```

Comment:
What is the newTable ? Is this a new mysql table with new schema for named-blob-update ?

Reply:
Yes, we created a new table with new schema (added 'version' and 'state' fields).